app/app/src/main/java/org/omecproject/up4/impl/Up4DeviceManager.java

```diff
@@ -1197,13 +1208,15 @@ private void checkStateAndReconcile() throws UpfProgrammableException {
                 Set<FlowRule> leaderRules =
                     StreamSupport.stream(flowRuleService.getFlowEntries(leaderUpfDevice).spliterator(), false)
                         .filter(r -> getLeaderUpfProgrammable().fromThisUpf(r))
+                        .filter(r -> r.state() == FlowEntryState.PENDING_ADD || r.state() == FlowEntryState.ADDED)
```

Comment:
Line 1211 & 1219 are needed.

Reply:
Otherwise we are coping unwanted rules (PENDING_DELETE, DELETED) to follower or vice versa.

Reply:
Can you submit this as a separate PR? We need to fix this on master as well.